docs/internal-architecture.md

```diff
@@ -0,0 +1,51 @@
+## Collector internal architecture
+There are a few resources available to understand how the collector works:
```

Comment:
It should be clear from context that we are talking about the Collector so I think we can remove it from the title. Also, let's be consistent with capitalization as well

```suggestion
## Internal architecture

There are a few resources available to understand how the Collector works:

```